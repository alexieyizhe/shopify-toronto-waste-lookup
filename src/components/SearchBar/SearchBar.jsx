import React from 'react';
import styled from 'styled-components';
import Search from 'react-feather/dist/icons/search';
import { ItemsContext } from '../../utils/siteContext';

import { searchBarEmptyPlaceholder } from '../../utils/siteData';
import { mediaSize } from '../../utils/siteTools';


const ComponentContainer = styled.div`
  height: 7vh;
  margin: 2vh 0; // margin allows for margin collapsing

  & > form {
    box-sizing: border-box;
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    height: 100%;
  }
`;

const SearchInputBox = styled.input`
  width: 90%;

  font-size: 1em;
  padding: 1vh;

  border: none; // css reset for default input styling
  border-radius: 5px;
  border: 1px solid grey;

  ${mediaSize.tablet`
    width: 80%;
  `}

  ${mediaSize.mobile`
    width: 70%;
  `}
`;

const SearchButton = styled.button`
  width: 7vh;

  background-color: ${props => props.theme.colors.secondary};

  border: none;
  border-radius: 5px;
`;

// Component
export default () => (
  <ItemsContext.Consumer>
    {({ searchQuery, updateSearch, startSearch }) => (
      <ComponentContainer>
        <form onSubmit={(e) => { e.preventDefault(); startSearch(); }} >
          <SearchInputBox placeholder={searchBarEmptyPlaceholder} value={searchQuery} onChange={(e) => updateSearch(e.target.value)} />
          <SearchButton><Search color="white" size="4vh" /></SearchButton>
        </form>
      </ComponentContainer>
    )}
  </ItemsContext.Consumer>
);
