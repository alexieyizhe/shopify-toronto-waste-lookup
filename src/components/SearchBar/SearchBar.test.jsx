import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from './SearchBar';

/* eslint-disable no-undef */
it('renders correctly', () => {
  const tree = renderer.create(<SearchBar />).toJSON();
  expect(tree).toMatchSnapshot();
});
