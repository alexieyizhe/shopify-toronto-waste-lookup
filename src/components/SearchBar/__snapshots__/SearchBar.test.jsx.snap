// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  height: 8vh;
  margin: 1vw 0;
}

.c0 > form {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c1 {
  width: 90%;
  padding: 1vw;
  font-size: 2vw;
  border: none;
  border-radius: 5px;
  border: 1px solid #333;
}

.c2 {
  width: 8vh;
  height: 8vh;
  background-color: #23975e;
  border-radius: 5px;
}

<div
  className="c0"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="c1"
      onChange={[Function]}
      placeholder="Enter a search"
    />
    <button
      className="c2"
    >
      <svg
        fill="none"
        height="2vw"
        stroke="white"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="2vw"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="11"
          cy="11"
          r="8"
        />
        <line
          x1="21"
          x2="16.65"
          y1="21"
          y2="16.65"
        />
      </svg>
    </button>
  </form>
</div>
`;
